test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,197 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayapiclientset "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:GatewayAPIController][Feature:Router][apigroup:gateway.networking.k8s.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithPodSecurityLevel("gatewayapi-controller", admissionapi.LevelBaseline)
+	)
+	const (
+		// The expected OSSM subscription name.
+		expectedSubscriptionName = "servicemeshoperator3"
+		// The expected OSSM operator namespace.
+		expectedSubscriptionNamespace = "openshift-operators"
+		// The gatewayclass name used to create ossm and other gateway api resources.
+		gatewayClassName = "openshift-default"
+		// gatewayClassControllerName is the name that must be used to create a supported gatewayClass.
+		gatewayClassControllerName = "openshift.io/gateway-controller/v1"
+	)
+	g.BeforeEach(func() {
```

Comment:
We need to check platformType and skip the tests if not on `AWSPlatformType, AzurePlatformType, GCPPlatformType`.